gigaword-indexer/src/main/java/edu/isi/vista/gigawordIndexer/AceDocument.java

```diff
@@ -0,0 +1,52 @@
+package edu.isi.vista.gigawordIndexer;
+
+import com.google.common.collect.AbstractIterator;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.Iterator;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+/**
+ * Single ACE Document as an article source
+ */
+public class AceDocument implements ArticleSource {
+    private final String docText;
+
+    private AceDocument(String fileText) {
+        this.docText = Objects.requireNonNull(fileText);
+    }
+
+    public static AceDocument AceDocumentFromPath(Path filePath) throws IOException {
+        return new AceDocument(new String(Files.readAllBytes(filePath)));
+    }
+
+    public Iterator<Article> iterator() {
+        return new ArticleIterator();
+    }
+
+    @Override
+    public void close() throws Exception
+    {
+
+    }
+
+    private class ArticleIterator extends AbstractIterator<Article> {
+        protected Article computeNext(){
+            // DOC ID marker
+            Pattern ACE_DOC_ID_PATTERN = Pattern.compile("<DOCID> (.*?) <.DOCID>");
+
+            Matcher m =
+                    ACE_DOC_ID_PATTERN.matcher(
+                            docText.substring(0, Math.min(120, docText.length())));
+            if (m.find()) {
+                String docId = m.group(1);
+                return new Article(docId, docText);
```

Comment:
This is going to feed all the markup directly into ElasticSearch, which will be confusing for the annotators and might mess up the HTML display. We need to clean it in a way which preserves offsets.

Is there documentation in the ACE corpus about how offsets are computed?

Reply:
I've been able to find that the XML tags are not counted and end-of-lines are counted as a single character. How should I proceed with the cleanup?

Reply:
Transform the original text which gets indexed by elastic search to remove all content in HTML tags and to collapse any Windows CRLF EOLs to `\n`.

Reply:
Added. I did this by removing all tags <.*?>.  however, one thing affected would be the source mentioned inside the tag. E.g: 
<DOCTYPE SOURCE="broadcast news"> NEWS STORY </DOCTYPE>
transforming into
NEWS STORY
Is this acceptable? (This appears uniformly mapped. I.e: all "broadcast news" have NEWS STORY)

Reply:
That is not a problem.

Reply:
Actually can you experiment with importing a few documents with the markup left in-place (but the newlines still fixed) into Inception and see how it appears in the interface? @elizlee can probably help you with this. If it isn't garbled, it may be best to leave the markup in place.